tests/IVIMmodels/unit_tests/test_ivim_fit_linear.py

```diff
@@ -45,3 +47,25 @@ def test_ivim_fit(f, D, Dp, bvals):
     npt.assert_allclose([f, D], [f_fit, D_fit], atol=1e-5)
     if not np.allclose(f, 0):
         npt.assert_allclose(Dp, Dp_fit, rtol=1e-2, atol=1e-3)
+
+def test_ivim_fit_saved(request):
+    file = pathlib.Path(request.node.fspath)
+    # print('current test file:', file)
+    generic = file.with_name('generic.json')
+    # print('current config file:', generic)
+    fit = LinearFit()
+    with generic.open() as f:
+        data = json.load(f)
+        bvals = data.pop('config')
+        bvals = bvals['bvalues']
+        for name, data in data.items():
+            print(f'{name} {data}')
+            signal = np.asarray(data['data'])
+            signal /= signal[0]
+            if data['f'] == 1.0:
```

Comment:
not sure why this is here. Would you not also need a if f==0 statement?

Reply:
That's really just because of my linear approximation fitting. I wouldn't expect real fitting algorithms to need it.

It's because it tries to fit non-flow data first and if it's all flow, it assumes that's non-flow and returns incorrect values. That's also why `f==0` is not needed. In that case it already greedily fit the non-flow and there's nothing left for flow to fit.